addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationAlertTest.java

```diff
@@ -0,0 +1,211 @@
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.automation;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.zaproxy.addon.automation.jobs.ActiveScanJob;
+
+public class AutomationAlertTest extends AbstractAutomationTest {
+    public final Integer scanRuleId;
+    public final String action;
+    public final String name;
+    public Pattern alertName;
+    public Pattern url;
+    public Pattern method;
+    public Pattern attack;
+    public Pattern param;
+    public Pattern evidence;
+    public Integer confidence;
+    public Integer risk;
+    public Pattern otherInfo;
+    public final String onFail;
+    public final String jobType;
+
+    public AutomationAlertTest(
+            Integer scanRuleId,
+            String action,
+            String name,
+            String alertName,
+            String url,
+            String method,
+            String attack,
+            String param,
+            String evidence,
+            String confidence,
+            String risk,
+            String otherInfo,
+            String onFail,
+            String jobType)
+            throws IllegalArgumentException {
+        super(name, onFail);
+
+        this.scanRuleId = scanRuleId;
+        this.action = action;
+        this.name = name;
+
+        if (!isNullOrEmpty(alertName)) {
+            this.alertName = Pattern.compile(alertName);
+        }
+
+        if (!isNullOrEmpty(url)) {
+            this.url = Pattern.compile(url);
+        }
+        if (!isNullOrEmpty(method)) {
+            this.method = Pattern.compile(method);
+        }
+        if (!isNullOrEmpty(attack)) {
+            this.attack = Pattern.compile(attack);
+        }
+        if (!isNullOrEmpty(param)) {
+            this.param = Pattern.compile(param);
+        }
+        if (!isNullOrEmpty(evidence)) {
+
+            this.evidence = Pattern.compile(evidence);
+        }
+        if (!isNullOrEmpty((confidence))) {
+            int idx = findIndexInArray(confidence, Alert.MSG_CONFIDENCE);
+            if (idx == -1) {
+                throw new IllegalArgumentException(
+                        Constant.messages.getString(
+                                "!automation.tests.alert.invalidConfidence!",
```

Comment:
The exclamation marks should not be used here - they are just used in unit tests where we know the message files dont get loaded correctly :)